packages/jaeger-ui/src/components/TracePage/TraceStatistics/index.tsx

```diff
@@ -392,79 +188,159 @@ export default class TraceStatistics extends Component<Props, State> {
+      {
+        title: 'Count',
+        dataIndex: 'count',
+        sorter: sorterFunction('count'),
+        defaultSortOrder: 'ascend',
+      },
+      {
+        title: 'Total',
+        dataIndex: 'total',
+        sorter: sorterFunction('total'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'Avg',
+        dataIndex: 'avg',
+        sorter: sorterFunction('avg'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'Min',
+        dataIndex: 'min',
+        sorter: sorterFunction('min'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'Max',
+        dataIndex: 'max',
+        sorter:sorterFunction('max'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Total',
+        dataIndex: 'selfTotal',
+        sorter:sorterFunction('selfTotal'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Avg',
+        dataIndex: 'selfAvg',
+        sorter: sorterFunction('selfAvg'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Min',
+        dataIndex: 'selfMin',
+        sorter:sorterFunction( 'selfMin' ),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Max',
+        dataIndex: 'selfMax',
+        sorter:sorterFunction('selfMax'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST in Duration',
+        dataIndex: 'percent',
+        sorter:sorterFunction('percent'),
+        render: (cell: string) => {
+          return `${cell} %`;
+        },
+      },
+    ];
+
+    columns = columns.map(val => {
+      return { ...val, onCell: record => onCellFunction(record) };
+    });
+
+    const findTablesAccToSelectors = (tableValue: ITableSpan[]): ITableSpan[] => {
+      const withDetail: ITableSpan[] = tableValue.filter((val: ITableSpan) => val.isDetail);
+      const withoutDetail: ITableSpan[] = tableValue.filter((val: ITableSpan) => !val.isDetail);
+      for (let i = 0; i < withoutDetail.length; i++) {
+        let newArr = withDetail.filter(value => value.parentElement === withoutDetail[i].name);
+        newArr = newArr.map((value, index) => {
+          const _key = {
+            key: `${i}-${index}`,
+          };
+          const value2 = { ...value, ..._key };
+          return value2;
+        });
+        const child = {
+          key: i.toString(),
+          children: newArr,
+        };
+        withoutDetail[i] = { ...withoutDetail[i], ...child };
+      }
+      return withoutDetail;
+    };
+    const withoutDetail: ITableSpan[] = findTablesAccToSelectors(this.state.tableValue);
```

Comment:
At this point why is withoutDetail an appropriate name?

Reply:
Like withDetail refers to the inner details rows or subgroups of each main row. So each main row does not have isDetail value we kept it as withoutDetail

Reply:
It would also be helpful to have a comment that defines an invariant about the data shape expectation. Say I have 2 groups, each has 3 subgroups. Does the final output structure has `2 + 2 * 3 = 8` entries, or only 2 entries each having 3 children? I found it very hard to deduce just from reading the code.